src/Models/ArduinoDeviceBase.ts

```diff
@@ -293,117 +133,103 @@ export abstract class ArduinoDeviceBase implements Device, LibraryManageable {
 
   abstract async preUploadAction(): Promise<boolean>;
 
-  abstract get version(): string;
-
   // Helper functions:
   generateCommonFiles(): void {

```

Comment:
> generateCommonFiles(): void { [](start = 1, length = 30)

Improvement: all generate*** functions should be asynchronous

Reply:
Done.